examples/NetCore2/ConsoleExample/Program.cs

```diff
@@ -41,23 +42,28 @@ private static IServiceProvider BuildDi()
         {
             var services = new ServiceCollection();
 
-            // Runner is the custom class
-            services.AddTransient<Runner>();
+            var config = new ConfigurationBuilder()
+                    .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                       .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+              .Build();
 
-            services.AddSingleton<ILoggerFactory, LoggerFactory>();
-            services.AddSingleton(typeof(ILogger<>), typeof(Logger<>));
-            services.AddLogging(builder => builder.SetMinimumLevel(LogLevel.Trace));
 
-            var serviceProvider = services.BuildServiceProvider();
-
-            var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+            services.AddSingleton<IConfiguration>(config);
+            // Runner is the custom class
+            services.AddTransient<Runner>();
 
-            // configure NLog
-            loggerFactory.AddNLog(new NLogProviderOptions
+            // Use Configure NLog and add NLog Factory
+            services.AddSingleton(new NLogProviderOptions
             {
                 CaptureMessageTemplates = true,
                 CaptureMessageProperties = true
             });
+            services.AddSingleton<ILoggerFactory, NLogLoggerFactory>();
+
+            services.AddSingleton(typeof(ILogger<>), typeof(Logger<>));
```

Comment:
The `AddLogging` already includes logic for registering `ILogger<>`, no need to have it twice